lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -176,9 +194,37 @@ def create_opensearch_users(
             logger.error(err)
             raise
 
+    def update_certs(self, relation_id, ca_chain=None):
+        """Update TLS certs passed into this relation.
+
+        If ca_chain is not provided, it'll get the app-admin CA generated by the TLS charm.
+        """
+        if not self.unit.is_leader():
+            # We're updating app databags in this function, so it can't be called on follower
+            # units.
+            return
+        if not ca_chain:
+            try:
+                ca_chain = self.charm.secrets.get_object(Scope.APP, CertType.APP_ADMIN.val).get(
+                    "chain"
+                )
+            except AttributeError:
+                # cert doesn't exist - presumably we don't yet have a TLS relation.
+                return
+        self.opensearch_provides.set_tls_ca(relation_id, "\n".join(ca_chain[::-1]))
+        self.opensearch_provides.set_tls(relation_id, "True")
+
+    def _on_relation_changed(self, event: RelationChangedEvent) -> None:
+        self.update_endpoints(event.relation)
+
     def _on_relation_departed(self, event: RelationDepartedEvent) -> None:
         """Check if this relation is being removed, and update the peer databag accordingly."""
-        if event.departing_unit == self.charm.unit:
+        # self.update_endpoints(
+        #     event.relation,
+        #     omit_endpoints={unit_ip(self.charm, event.departing_unit, PeerRelationName)},
+        # )
```

Comment:
**question, todo** is there a reason to keep this in? Otherwise we shuold rather remove commented code in general

Reply:
nope, removed